src/components/signup/SmsForm.js

```diff
@@ -184,17 +184,35 @@ const SMSAction = ({
   tries,
   styles,
 }) => {
+  const smsRateLimitNormalized = smsRateLimit / 1000
   const [showWait, setWait] = useState(true)
+  const [waitTime, setWaitTime] = useState(smsRateLimitNormalized)
   const [isCall, setIsCall] = useState(false)
   const _handleRetry = useOnPress(handleRetry)
   const _handleRetryWithCall = useOnPress(handleRetryWithCall)
   const _handleSkip = useOnPress(handleSkip)
 
+  let x
```

Comment:
1. **should** use ref instead, could have unexpected behaviour as the x variable redefines each render.
```suggestion
  const xRef = useRef()
```
2. maybe to move this effect & state vars related to the separate hook e.g. useWaitTime()